shared/us-east-1/tools-costs-notifications/src/script.py

```diff
@@ -0,0 +1,338 @@
+    tag_headers = ' '.join(f'<th style="width: 12%; padding: 10px; text-align: right;">Tag {tag_key}</th>' for tag_key in tags)
+
+    html_table += f"""<table border='1' style='width: 100%; border-collapse: collapse;'>
+                        <thead>
+                            <tr style='background-color: #f2f2f2;'>
+                                <th style='width: 40%; padding: 10px; text-align: left;'>Service</th>
+                                <th style='width: 12.5%; padding: 10px; text-align: right;'>Cost on {last_month_name}</th>
+                                <th style='width: 12.5%; padding: 10px; text-align: right;'>Cost on Previous Month</th>
+                                <th style='width: 12.5%; padding: 10px; text-align: right;'>Variation %</th>
+                                {tag_headers}
+                            </tr>
+                        </thead><tbody>"""
+
+    total_cost = Decimal(0)
+    total_prev_month_cost = Decimal(0)
+    total_tag_costs = {tag_key: Decimal(0) for tag_key in tags}
+
+    for group in cost_data:
+        service = group['Keys'][0]
+        current_cost = Decimal(group['Metrics']['UnblendedCost']['Amount']).quantize(Decimal('0.00'))  # Round to 2 decimals
+        prev_month_cost = prev_cost_data_dict.get(service, Decimal(0)).quantize(Decimal('0.00'))  # Round to 2 decimals
+
+        variation_percent = calculate_variation(prev_month_cost, current_cost)
+        variation_color = 'green' if variation_percent > 0 else 'red'
+
+        # Initialize the row with service name, current cost, previous month cost, and variation
+        row = [f'<td>{service}</td>',
+               f'<td style="text-align:right;">${current_cost:.2f}</td>',
+               f'<td style="text-align:right;">${prev_month_cost:.2f}</td>',
+               f'<td style="text-align:right; color:{variation_color};">{variation_percent:.2f}%</td>']
+        
+        # Add cost columns for each tag, or an empty cell if the tag doesn't exist for this service
+        for tag_key, tag_value in tags.items():
+            tag_cost = get_tag_cost(ce_client, ACCOUNTS[account_name]['id'], tag_key, tag_value, start_date_str, end_date_str)
+            tag_cost_amount = sum(Decimal(group['Metrics']['UnblendedCost']['Amount']) for group in tag_cost)
+            row.append(f'<td style="text-align:right;">${tag_cost_amount:.2f}</td>')
+            total_tag_costs[tag_key] += tag_cost_amount
+
+        # Ensure that each row has the same number of columns
+        while len(row) < len(tags) + 4:
+            row.append('<td></td>')
+
+        html_table += '<tr>' + ' '.join(row) + '</tr>'
+
+        total_cost += current_cost
+        total_prev_month_cost += prev_month_cost
+
+
+    variation_percent_month_to_month = calculate_variation(total_prev_month_cost, total_cost)
+    variation_color_month_to_month = 'green' if variation_percent_month_to_month > 0 else 'red'
```

Comment:
a dummy comment but why green if it raise? it is a bad thing :stuck_out_tongue: 
shouldn't be the other way around?